decoder.go

```diff
@@ -404,16 +404,20 @@ func (d *decoder) decodeMap(name string, node ast.Node, result reflect.Value) er
 }
 
 func (d *decoder) decodePtr(name string, node ast.Node, result reflect.Value) error {
-	// Create an element of the concrete (non pointer) type and decode
-	// into that. Then set the value of the pointer to this type.
+	// lookup the concrete (non pointer) type and decode into that
+	// if the pointer is nil, create an element of the type and set the pointer to that
```

Comment:
decodePtr overall exists to run decode on the reflect.Value wrapped around a concrete instance of the type pointed to. It's just reflect.Indirect, except that it may have to create the type. I'll take another swing at the comment.